src/faldbt/project.py

```diff
@@ -86,45 +91,81 @@ def _set_status(self, status: str):
 
 @dataclass
 class DbtTest(_DbtNode):
-    column: str = field(init=False)
-    source: Optional[Tuple[str, str]] = field(init=False, default=None)
-    model: Optional[str] = field(init=False, default=None)
+    model_ids: List[str] = field(init=False, default_factory=list)
+    source_ids: List[str] = field(init=False, default_factory=list)
 
-    def __post_init__(self):
-        node = self.node
+    @classmethod
+    def init(cls, node):
         if node.resource_type == NodeType.Test:
-            if hasattr(node, "test_metadata"):
-                metadata: TestMetadata = self.node.test_metadata
+            if isinstance(node, ParsedGenericTestNode):
+                test = DbtGenericTest(node=node)
+            elif isinstance(node, ParsedSingularTestNode):
+                test = DbtSingularTest(node=node)
+            else:
+                raise ValueError(f"Unexpected test class {node.__class__.__name__}")
 
-                # metadata.kwargs looks like this:
-                # kwargs={'column_name': 'y', 'model': "{{ get_where_subquery(ref('boston')) }}"}
-                # and we want to get 'boston' is the model name that we want extract
-                # except in dbt v 0.20.1, where we need to filter 'where' string out
-                self.column = metadata.kwargs.get("column_name", None)
+            for dep in test.node.depends_on.nodes:
+                if dep.startswith("model."):
+                    test.model_ids.append(dep)
+                if dep.startswith("source."):
+                    test.source_ids.append(dep)
 
-                # TODO: Handle package models?
-                refs = re.findall(r"ref\('([^']+)'\)", metadata.kwargs["model"])
+            return test
+        else:
+            raise TypeError(
+                f"Initialized DbtTest with node of type {node.resource_type}"
+            )
 
-                if refs:
-                    self.model = refs[0]
 
-                sources = re.findall(
-                    r"source\('([^']+)', *'([^']+)'\)", metadata.kwargs["model"]
-                )
-                if sources:
-                    self.source = sources[0]
-            else:
-                logger.warn(f"Non-generic test was not processed: {node.name}")
+@dataclass
+class DbtGenericTest(DbtTest):
+    column: Optional[str] = field(init=False)
+
+    def __post_init__(self):
+        assert isinstance(self.node, ParsedGenericTestNode)
+        self.column = self.node.column_name
+
+    @property
+    def source_id(self):
+        if self.source_ids:
+            return self.source_ids[0]
+
+    @property
+    def model_id(self):
+        if self.model_ids:
+            return self.model_ids[0]
+
+    # TODO: Deprecate?
+    @property
+    def source(self):
+        if self.source_id:
+            parts = self.source_id.split(".")
+            return parts[-2], parts[-1]
+
+    # TODO: Deprecate?
```

Comment:
I'd say yes, the model_id/source_id APIs seems enough. All the post processing can go to client code. Though no strong opinions, it is very light so it can stay either.

Reply:
We can solve it later, it is lightweight.

Reply:
Yep!